src/main/java/org/jenkinsci/plugins/codesonar/services/HttpServiceRequest.java

```diff
@@ -0,0 +1,57 @@
+package org.jenkinsci.plugins.codesonar.services;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Map;
+
+import org.jenkinsci.plugins.codesonar.CodeSonarPluginException;
+
+
+/** Provides extended properties for a CodeSonar hub request. */
+public class HttpServiceRequest {
+    private URI uri;
+    private String uriString;
+    private Collection<Map.Entry<String,String>> headers;
+
+    public HttpServiceRequest(URI uri) throws CodeSonarPluginException {
+        if (uri == null) {
+            throw new CodeSonarPluginException("URI cannot be null");
+        }
+        this.uri = uri;
+        this.uriString = uri.toString();
```

Comment:
Add a TODO to remove the parallel storage of URI object and URI string.  What if they got out of sync with totally different URLs?

Reply:
Done.  Make class members `final`.